marshal/consumer.go

```diff
@@ -132,9 +138,18 @@ func (m *Marshaler) NewConsumer(topicNames []string, options ConsumerOptions) (*
 					}
 					c.claims[topic][partID] = newClaim(
 						topic, partID, c.marshal, c.messages, options)
+
+					// update topic claims
+					if options.ClaimEntireTopic && partID == 0 {
+						c.claimedTopics[topic] = true
+					}
 				}
 			}
 		}
+		// send topic claim notification
+		if options.ClaimEntireTopic {
+			c.topicClaimsChan <- c.claimedTopics
```

Comment:
You're right about the reference...I forgot to copy here. I copied in the other place :).
This should never block because it's where the channel is created before it's returned and it has a size of 1